Add OpenAPI Spec, Generate Client Code and Sanitations (#2)
manodyaSenevirathne/module-ballerinax-openai-chat: docs/spec/sanitations.md

_Authors_: @manodyaSenevirathne \
_Created_: 2024/08/05 \
_Updated_: 2024/08/05 \
_Edition_: Swan Lake

# Sanitation for OpenAPI specification

This document records the sanitation done on top of the official OpenAPI specification from OpenAI Finetune. The OpenAPI specification is obtained from the [OpenAPI specification for the OpenAI API](https://github.com/openai/openai-openapi/blob/master/openapi.yaml). These changes are implemented to enhance the overall usability and readability of the generated client.

> 1. **Removed the `Deprecated: true` property of certain schemas**:

Reviewer: ```suggestion
1. **Removed the `Deprecated: true` property from the below schemas**:
```

   - **Original**:
      - Deprecated: `true`

   - **Updated**:
      - Removed the `deprecated` parameter

   - **Reasons**: The original configuration was generated successfully, but it led to compile-time warnings. To ensure smoother compilation and avoid potential issues, the configuration has been revised.

2. **Removed the `default:null` property of certain schemas**:

   - **Changed Schemas**: `CreateCompletionRequest`,`ChatCompletionStreamOptions`,`CreateChatCompletionRequest`

   - **Original**:

>       - defaullt: `null`

Reviewer: ```suggestion
      - default: `null`
```
